File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.py
from itertools import permutations #순열

def prime(x):
    if x < 2:
        return False
    for i in range(2,x):
        if x % i == 0:
            return False
    return True

def solution(numbers):
    answer = 0
    allNums = set()
    li = list(numbers)
    for i in range(1, len(numbers)+1):
        permLi = permutations(li, i)
        for perm in permLi:
            num = int(''.join(perm))
            allNums.add(num)

    for num in allNums:
        if prime(num):
            answer += 1
    return answer

File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.java
import java.util.*;

class Solution {
    static Set<Integer> numberSet = new HashSet<>();
    
    public int solution(String numbers) {
        int answer = 0;
        boolean[] visited = new boolean[numbers.length()];
        create("", numbers, visited);
        
        for (int num : numberSet) {
            if (isPrime(num)) {
                answer++;
            }
        }
        
        return answer;
    }
    
    public static void create(String current, String numbers, boolean[] visited) {
        if (!current.equals("")) {
            numberSet.add(Integer.parseInt(current));
        }
        
        for (int i = 0; i < numbers.length(); i++) {
            if (!visited[i]) {
                visited[i] = true;
                create(current + numbers.charAt(i), numbers, visited);
                visited[i] = false;
            }
        }
    }
    
    public static boolean isPrime(int num) {
        if (num < 2) return false;
        for (int i = 2; i <= Math.sqrt(num); i++) {
            if (num % i == 0) return false;
        }
        
        return true;
    }
}